pkg/controller/machine.go

```diff
@@ -358,10 +358,12 @@ func (c *controller) machineCreate(machine *v1alpha1.Machine, driver driver.Driv
 	glog.V(2).Infof("Created machine: %q, MachineID: %s", machine.Name, actualProviderID)
 
 	for {
+		machineName := machine.Name
 		// Get the latest version of the machine so that we can avoid conflicts
 		machine, err := c.controlMachineClient.Machines(machine.Namespace).Get(machine.Name, metav1.GetOptions{})
 		if err != nil {
-			return err
+			glog.Warningf("Machine GET failed for %q. Retrying, error: %s", machineName, err)
+			continue
```

Comment:
Would [`k8s.io/client-go/util/retry`](https://godoc.org/k8s.io/client-go/util/retry) be of use here?

Reply:
Yes, @amshuman-kr. The retry loop could be used here. However, Martin had mentioned that we should avoid loops like these in the controller and should return an error and retry the entire reconciliation from scratch. I plan to get rid of all these loops once the OOT is over. So this change is to fix any issues only until then. 

So I could introduce the retry loop but then we would be getting rid of it in the future. It's your call.